identity_stardust/src/client/client_ext.rs

```diff
@@ -0,0 +1,484 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::borrow::Cow;
+use std::ops::Deref;
+use std::str::FromStr;
+
+use iota_client::api_types::responses::OutputResponse;
+use iota_client::block::address::Address;
+use iota_client::block::output::feature::IssuerFeature;
+use iota_client::block::output::feature::SenderFeature;
+use iota_client::block::output::unlock_condition::AddressUnlockCondition;
+use iota_client::block::output::unlock_condition::GovernorAddressUnlockCondition;
+use iota_client::block::output::unlock_condition::StateControllerAddressUnlockCondition;
+use iota_client::block::output::AliasId;
+use iota_client::block::output::AliasOutput;
+use iota_client::block::output::AliasOutputBuilder;
+use iota_client::block::output::BasicOutputBuilder;
+use iota_client::block::output::Feature;
+use iota_client::block::output::Output;
+use iota_client::block::output::OutputId;
+use iota_client::block::output::RentStructure;
+use iota_client::block::output::UnlockCondition;
+use iota_client::block::payload::transaction::TransactionEssence;
+use iota_client::block::payload::Payload;
+use iota_client::block::Block;
+use iota_client::secret::SecretManager;
+use iota_client::Client;
+
+use crate::error::Result;
+use crate::Error;
+use crate::NetworkName;
+use crate::StardustDID;
+use crate::StardustDocument;
+
+/// An extension trait for a [`Client`] that provides helper functions publication and resolution of
+/// DID documents in Alias Outputs.
```

Comment:
Perhaps it should be mentioned that this trait is mainly meant to be used rather than implemented?